scio-examples/src/main/scala/com/spotify/scio/examples/extra/ParquetExample.scala

```diff
@@ -0,0 +1,138 @@
+package com.spotify.scio.examples.extra
+
+import com.spotify.scio._
+import com.spotify.scio.parquet.avro._
+import com.spotify.scio.parquet.types._
+import com.spotify.scio.values.SCollection
+import com.spotify.scio.avro.Account
+import com.spotify.scio.io.ClosedTap
+import org.apache.hadoop.conf.Configuration
+import org.apache.avro.generic.GenericRecord
+
+object ParquetExample {
+
+  case class AccountInput(id: Int, `type`: String, name: String, amount: Double)
+
+  case class AccountOutput(id: Int, name: String)
+
+  // See this link for parquet writer tuning https://spotify.github.io/scio/io/Parquet.html#performance-tuning
+  lazy val fineTunedParquetWriterConfig: Configuration = {
+    val conf: Configuration = new Configuration()
+    conf.setInt("parquet.block.size", 1073741824) // 1 * 1024 * 1024 * 1024 = 1 GiB
+    conf.set("fs.gs.inputstream.fadvise", "RANDOM")
+    conf
+  }
+
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+
+    val m = args("method")
+    m match {
+      // Read Parquet files as Specific Avro Records
+      case "avroSpecificIn" => avroSpecificIn(sc, args)
+
+      // Read Parquet files as Generic Avro Records
+      case "avroGenericIn" => avroGenericIn(sc, args)
+
+      // Read Parquet files as Scala Case Classes
+      case "typedIn" => typedIn(sc, args)
+
+      // Write dummy Parquet Avro records
+      case "avroOut" => avroOut(sc, args)
+
+      // Write dummy Parquet records using case classes
+      case "typedOut" => typedOut(sc, args)
+
+      case _ => throw new RuntimeException(s"Invalid method $m")
+    }
+
+    sc.run().waitUntilDone()
+    ()
+  }
+
+  private def avroSpecificIn(sc: ScioContext, args: Args): ClosedTap[String] = {
+
+    // Account is an Avro Specific record. The avsc schema for the same can be found in scio-schemas/src/main/avro/schema.avsc
+    // Macros for generating column projections and row predicates
+    val projection =
+      Projection[Account](_.getId, _.getName, _.getAmount) // Only these columns will be projected
+    val predicate = Predicate[Account](x =>
+      x.getAmount > 0
+    ) // Will skip row groups where this column value check is not satisfied
+
+    sc.parquetAvroFile[Account](args("input"), projection, predicate)
+      // The result Account records are not complete Avro objects. Only the projected columns are present while the rest are null.
+      // These objects may fail serialization and it’s recommended that you map them out to tuples or case classes right after reading.
+      .map(x => AccountOutput(x.getId, x.getName.toString))
+      .map(_.toString)
+      .saveAsTextFile(args("output"))
+  }
+
+  private def avroGenericIn(sc: ScioContext, args: Args): ClosedTap[String] =
+    // Now the fields in Account's schema act as our projection
+    sc.parquetAvroFile[GenericRecord](args("input"), Account.getClassSchema)
+
+      // GenericRecord objects are also not Serializable.
```

Comment:
this statement isn't true--`GenericRecord`s as a class _are_ serializable it can just be inefficient (because the `schema` [needs to be ser/de-ed as well](https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/coders/instances/AvroCoders.scala).

Reply:
True, My bad. I confused it with Avro's `Schema` class that is not serializable (unless converted to a String).
I will get rid of this line 👍